File: Java_Programming/Competitive/Assignment_34.java
// Program which accept one number from user and toggle 7th and 10th bit of that number. Return modified number.
//
// Input: 137
// Output: 713


import java.util.Scanner;

class Bitwise
{

    // Position : 7 & 10
    public int ToggleBit(int iNo)
    {
        int iMask = 0x00000240;

        iNo = iMask ^ iNo;
        
        return iNo;
    }
}

class Assignment_34
{

    public static void main(String A[])
    {
        Bitwise bobj = new Bitwise();
        Scanner sobj = new Scanner(System.in);
        int iValue = 0, iRet = 0;

        System.out.println("Enter number : ");
        iValue = sobj.nextInt();

        iRet = bobj.ToggleBit(iValue);

        System.out.println("Modified number : "+iRet);
    }

}

File: C_Programming/Competitive/Assignment_34.c
// Program to find odd factorial of given number.
//
// Input: 5
// Output: 15    (5 * 3 * 1)
//
// Input: -5
// Output: 15    (5 * 3 *  1)
//
// Input: 10
// Output: 945 (9 * 7 * 5 * 3 * 1)


#include <stdio.h>

int OddFactorial(int iNo)
{
    int iCnt = 0;
    int iMul = 0;

    if(iNo < 0)
    {
        iNo = -iNo;
    }

    iMul = 1;
    for(iCnt = iNo;iCnt >= 2;iCnt--)
    {
        if((iCnt % 2) != 0)
        {
            iMul = iMul * iCnt;
        }
    }

    return iMul;
}

int main()
{
    int iValue = 0;
    int iRet = 0;

    printf("Enter number : ");
    scanf("%d",&iValue);

    iRet = OddFactorial(iValue);

    printf("Odd Factorial of number is %d\n",iRet);
    
    return 0;
}

// Time Complexity : O(N)
// Where N > 0